Tools/LyTestTools/ly_test_tools/o3de/asset_processor.py

```diff
@@ -263,42 +263,70 @@ def _attempt_connection():
         waiter.wait_for(_attempt_connection, timeout=timeout, exc=err)
         return True, None
 
-    def stop(self, timeout=60):
+    def stop(self, timeout=60, ap_stop_test=False):
         """
          Stops the AssetProcessor. First attempts a clean shutdown over the control channel
          if open.  Calls terminate if no control connection is open
          :param timeout: How long to wait, in seconds, for AP to shut down after receiving quit message.
             This timeout is a longer default because AP doesn't quit immediately on receiving the quit message,
             it waits until finishing its current task, which can sometimes take a while.
+         :param ap_stop_test: Informs the stop function whether or not the stop is intentional for testing purposes.
+            If set to true, any terminate call within the function will return a ValueError with a specific string
+            for the tests to compare against.
          :return: None
          """
         if not self._ap_proc:
             logger.warning("Attempting to quit AP but none running")
             return
 
         if not self._control_connection:
-            logger.info("No control connection open, using terminate")
-            return self.terminate()
+            if ap_stop_test:
+                logger.info("No control connection open, using terminate")
+                self.terminate()
+                raise ValueError("NoControlConnection")
+            else:
+                logger.info("No control connection open, using terminate")
+                return self.terminate()
```

Comment:
I think it might be better to just set a return value and return that at the end, this way there isn't the chance of different code running for the test vs not.  Plus it'll be cleaner since you don't need all the if statements.  I'd add an enum with the different return values:

```
from enum import Enum

class StopReason(Enum):
    NO_CONTROL = 1
    IO_ERROR = 2
```

etc, etc...

```suggestion
                logger.info("No control connection open, using terminate")
                self.terminate()
                return StopReason.NO_CONTROL
```